geometry/meshop.cpp

```diff
@@ -252,6 +252,118 @@ void saveAsPly( const Mesh &mesh, const boost::filesystem::path &filepath){
     }
 }
 
+void saveAsPlyWithFeatures(const boost::filesystem::path& filepath,
+                           const Mesh& mesh,
+                           const math::Points3i& vertexColors,
+                           const math::Points3i& faceColors,
+                           const math::Points3& vertexNormals,
+                           const std::vector<int>& faceLabels)
+{
+    LOG(info2) << "Saving mesh to file <" << filepath << ">.";
+
+    std::ofstream out(filepath.string().c_str());
+    out.setf(std::ios::scientific, std::ios::floatfield);
```

Comment:
A good practice is to either check the state of the file stream or set an exception mask so in case of an error the job never finishes silently.

Reply:
Again, followed `saveAsPly()` too blindly, added the exceptions there as well.